integration_tests/README.md

```diff
@@ -303,3 +303,66 @@ The marks you care about are all in marks.py
 
 For the most part you can ignore this file. It provides the underlying Spark session to operations that need it, but most tests should interact with
 it through `asserts.py`.
+
+## Guidelines for Testing
+
+When support for a new operator is added to `rapids-plugin-4-spark`, it is recommended that
+the following conditions be covered in its corresponding integration tests:
+
+### 1. Cover all supported data types
+Ensure that tests cover all data types supported by the added operation. An exhaustive list of data types supported in 
+Spark is available [here](https://spark.apache.org/docs/latest/sql-ref-datatypes.html). These include:
+   * Numeric Types (`ByteType`, `ShortType`, `IntegerType`, etc.)
+   * Strings (`StringType`, `VarcharType`)
+   * Booleans (`BooleanType`)
+   * Chrono Types (`TimestampType`, `DateType`, `Interval`)
+   * Complex Types (`ArrayType`, `StructType`, `MapType`)
+
+`data_gen.py` provides `DataGen` classes that help generate test data in integration tests.
+
+### 2. Nested Data types
+Complex data types (`ArrayType`, `StructType`, `MapType`) warrant extensive testing for various combinations of nesting.
```

Comment:
This is a good point. We need a way for us to create nesting in our integration tests quickly instead of typing it out but at the same time be careful the nesting isn't too deep as it slows down the tests.

Reply:
I have created a follow-on issue for it. https://github.com/NVIDIA/spark-rapids/issues/2759